src/Compiler/TypedTree/tainted.fs

```diff
@@ -101,11 +101,12 @@ type internal Tainted<'T> (context: TaintedContext, value: 'T) =
             | :? TypeProviderError -> reraise()
             | :? AggregateException as ae ->
                     let errNum,_ = FSComp.SR.etProviderError("", "")
-                    let messages = [for e in ae.InnerExceptions -> e.Message]
+                    let messages = [for e in ae.InnerExceptions -> if isNull e.InnerException then e.Message else (e.Message + ": " + e.GetBaseException().Message)]
                     raise <| TypeProviderError(errNum, this.TypeProviderDesignation, range, messages)
             | e -> 
                     let errNum,_ = FSComp.SR.etProviderError("", "")
-                    raise <| TypeProviderError((errNum, e.Message), this.TypeProviderDesignation, range)
+                    let error = if isNull e.InnerException then e.Message else (e.Message + ": " + e.GetBaseException().Message)
```

Comment:
As we discussed and as explained in this comment (https://github.com/dotnet/fsharp/pull/17251#issuecomment-2132294934), we cannot rely on the virtual method `GetBaseException()`. Use the code snippet from that comment (copied here for ease), or something similar instead, to prevent this line causing spurious issues.

```f#
[<TailCall>]
let rec getMostInnerException (ex: exn) =
    if isNull ex then 
        // should never happen, we are in a try-with (we may consider throwing instead of returning?)
        ArgumentNullException("Unreachable code. Caught exception is null", nameof ex) :> exn
    else
        if isNull ex.InnerException then ex
        else getMostInnerException ex.InnerException
```

```suggestion
                    let error = getMostInnerException e
```